test/JDBC/expected/test_dynamic_local_vars.out

```diff
@@ -0,0 +1,1303 @@
+
+
+declare @v varchar(20) = "ABC", @v2 varchar(20)="XYZ";
+select @v += reverse("a""b''c'd"), @v2 += @v + reverse("x""y''z");
+select @v, @v2
+GO
+~~START~~
+varchar#!#varchar
+ABCd'c''b"a#!#XYZABCd'c''b"az''y"x
+~~END~~
+
+
+declare @v varchar(20) = "ABC", @v2 varchar(20)="XYZ";
+select @v += reverse("a""b''c'd"), @v2 += REVERSE( @v + reverse("x""y''z"));
+select @v, @v2
+GO
+~~START~~
+varchar#!#varchar
+ABCd'c''b"a#!#XYZx"y''za"b''c'dCBA
+~~END~~
+
+
+SET QUOTED_IDENTIFIER ON
+GO
+
+declare @v varchar(20) = 'ABC', @v2 varchar(20)='XYZ';
+select @v += 'abc', @v2 += 'xyz';
+select @v, @v2
+GO
+~~START~~
+varchar#!#varchar
+ABCabc#!#XYZxyz
+~~END~~
+
+
+declare @a int = 1, @b int = 2;
+select @a = 2, @b = @a + 2
+select @a, @b
+GO
+~~START~~
+int#!#int
+2#!#4
+~~END~~
+
+
+declare @a int = 1, @b int = 2;
+select @a += 2, @b -= @a + 2
+select @a, @b
+GO
+~~START~~
+int#!#int
+3#!#-3
+~~END~~
+
+
+-- xml methods
+DECLARE @a bit = 1
+DECLARE @xml XML = '<artists> <artist name="John Doe"/> <artist name="Edward Poe"/> <artist name="Mark The Great"/> </artists>'
+SELECT @a |= @xml.exist('/artists/artist/@name')
+select @a
+GO
+~~START~~
+bit
+1
+~~END~~
+
+
+DECLARE @a bit = 1
+DECLARE @xml XML;
+SELECT @xml  = '<artists> <artist name="John Doe"/> <artist name="Edward Poe"/> <artist name="Mark The Great"/> </artists>', @a |= @xml.exist('/artists/artist/@name')
+select @a
+GO
+~~START~~
+bit
+1
+~~END~~
+
+
+-- test all kind of udts
+create type udt from NCHAR
+go
+
+declare @a udt
+select @a = 'anc'
+select @a
+GO
+~~START~~
+nchar
+a
+~~END~~
+
+
+DROP type udt 
+GO
+
+create type varchar_max from varchar(max)
+GO
+
+DECLARE @a varchar_max
+SELECT @a = '12345678901234567890123456789012345';
+SELECT LEN(@a), DATALENGTH(@a)
+SELECT @a
+GO
+~~START~~
+int#!#int
+35#!#35
+~~END~~
+
+~~START~~
+varchar
+12345678901234567890123456789012345
+~~END~~
+
+
+DROP type varchar_max
+GO
+
+create type num_def from numeric
+GO
+
+declare @a numeric;
+declare @b num_def;
+SET @a=100.41;
+SET @b=200.82;
+SELECT @a, @b
+select @a+@b as r;
+GO
+~~START~~
+numeric#!#numeric
+100#!#201
+~~END~~
+
+~~START~~
+numeric
+301
+~~END~~
+
+
+drop type num_def
+GO
+
+/*
+ * UPDATE test
+ */
+create table local_var_tst (id int) 
+GO
+
+TRUNCATE table local_var_tst
+GO
+
+insert into local_var_tst values (1)
+insert into local_var_tst values (2)
+insert into local_var_tst values (6)
+GO
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+
+
+
+
+
+-- declare @i int 
+-- update local_var_tst set id =10, @i = id * 5
+-- select @i
+-- GO
+-- select * from local_var_tst;
+-- GO
+-- -- need to fix this
+-- declare @i int 
+-- set @i = 0
+-- update local_var_tst set id = @i, @i = id * 5
+-- select @i
+-- GO
+-- select * from local_var_tst;
+-- GO
+-- txn does not affect local variables
+begin tran
+declare @i int 
+update local_var_tst set id = 5, @i = id * 5
+select @i
+ROLLBACK tran
+select @i
+GO
+~~ROW COUNT: 3~~
+
+~~START~~
+int
+25
+~~END~~
+
+~~START~~
+int
+25
```

Comment:
30 as well here

Reply:
will be fixed with https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/3028